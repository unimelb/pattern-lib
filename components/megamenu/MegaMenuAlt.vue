<template>
  <header class="mega-menu-alt">
    <a
      href="#main"
      class="mega-menu-alt__skip-link screen-reader-jump-to__skippy screenreaders-only screenreaders-only-focusable btn--inverted btn btn--xsml">
      Skip to content
    </a>
    <div class="mega-menu-alt__inner bg-inverted">
      <Logo size="md" />
      <div class="mega-menu-alt__menu">
        <div class="mega-menu-alt__menu-top">
          <nav aria-label="Quick links">
            <ul class="mega-menu-alt__secondary-links list-reset">
              <li
                v-for="(link, index) in secondaryLinks"
                :key="index">
                <a
                  :href="link.href"
                  class="link-reset">
                  {{ link.text }}
                </a>
              </li>
            </ul>
          </nav>
          <ul class="mega-menu-alt__actions list-reset">
            <li class="mega-menu-alt__contact">
              <a
                href="https://www.unimelb.edu.au/contact"
                class="link-reset">
                Contact
              </a>
            </li>
            <li class="mega-menu-alt__search">
              <button
                ref="searchBtn"
                :aria-expanded="String(isSearchOpen)"
                class="link-reset"
                @click="openSearch">
                Search
                <svg
                  focusable="false"
                  role="presentation">
                  <use xlink:href="#icon-search" />
                </svg>
              </button>
            </li>
            <li class="mega-menu-alt__mobile-menu">
              <button
                ref="menuBtn"
                :aria-expanded="String(isMobileOpen)"
                class="link-reset"
                @click="openMobileMenu">
                Menu
                <svg
                  focusable="false"
                  aria-label="icon menu"
                  viewBox="10 10 26 28">
                  <path d="M6 36h36v-4H6v4zm0-10h36v-4H6v4zm0-14v4h36v-4H6z" />
                </svg>
              </button>
            </li>
          </ul>
        </div>
        <nav aria-label="Site">
          <ul class="mega-menu-alt__primary-links list-reset">
            <li
              v-for="(link, index) in primaryLinks"
              :key="index">
              <a
                :href="link.href"
                class="link-reset">
                {{ link.text }}
              </a>
            </li>
          </ul>
        </nav>
        <MegaMenuAltMobile
          v-model="isMobileOpen"
          :primary-links="primaryLinks"
          :secondary-links="secondaryLinks"
          @close="closeMobileMenu" />
        <div
          class="page-header-search"
          :class="{ active: isSearchOpen }"
          @keydown.esc="closeSearch">
          <Trap :disabled="!isSearchOpen">
            <PageSearchForm>
              <a
                href="#"
                aria-haspopup="true"
                class="page-header-search__close link-reset"
                @click.prevent="closeSearch">
                <svg
                  focusable="false"
                  role="presentation">
                  <use xlink:href="#icon-close" />
                </svg>
                Close
              </a>
            </PageSearchForm>
          </Trap>
        </div>
      </div>
    </div>
  </header>
</template>

<script>
import Trap from 'vue-focus-lock';
import PageSearchForm from 'components/search/PageSearchForm.vue';
import Logo from 'components/logo/Logo.vue';
import MegaMenuAltMobile from './MegaMenuAltMobile.vue';

export default {
  components: {
    Logo,
    MegaMenuAltMobile,
    PageSearchForm,
    Trap,
  },
  data() {
    return {
      isMobileOpen: false,
      isSearchOpen: false,
      primaryLinks: [
        { text: 'Home', href: '/home-page' },
        { text: 'Study at Melbourne', href: 'https://study.unimelb.edu.au/' },
        { text: 'Research and innovation', href: 'https://research.unimelb.edu.au/' },
        { text: 'About us', href: 'https://about.unimelb.edu.au/' },
        { text: 'Virtual Campus', href: 'https://study.unimelb.edu.au/discover/virtual-tour' },
      ],
      secondaryLinks: [
        { text: 'Current students', href: 'https://students.unimelb.edu.au/' },
        { text: 'Staff', href: 'https://staff.unimelb.edu.au/' },
        { text: 'Library', href: 'https://library.unimelb.edu.au/' },
      ],
    };
  },
  computed: {
    mobileMenuLinks() {
      return [
        ...this.primaryLinks,
        ...this.secondaryLinks,
        { text: 'Contact', href: 'https://www.unimelb.edu.au/contact' },
      ];
    },
  },
  methods: {
    openMobileMenu() {
      this.isMobileOpen = true;
    },
    openSearch() {
      this.isSearchOpen = true;
    },
    closeMobileMenu() {
      this.isMobileOpen = false;
      this.$nextTick(() => {
        this.$refs.menuBtn.focus();
      });
    },
    closeSearch() {
      this.isSearchOpen = false;
      this.$nextTick(() => {
        this.$refs.searchBtn.focus();
      });
    },
  },
};
</script>
