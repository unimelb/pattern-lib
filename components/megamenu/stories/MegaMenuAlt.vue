<template>
  <div>
    <MegaMenuAlt
      :primary-links="primaryLinks"
      :secondary-links="secondaryLinks" />
    <main
      id="main"
      role="main">
      <section-wrap>
        <h1>Main content area</h1>
        <a href="#">Anchor Link</a>
        <p>
          Sit dolor sint occaecat nulla. Excepteur sit excepteur ex laborum adipisicing est in aliquip ex. Velit reprehenderit labore dolore ipsum officia in do do aute commodo enim proident id ut. Exercitation eiusmod excepteur enim tempor anim eiusmod eiusmod ad aute pariatur culpa. Nostrud fugiat adipisicing deserunt ut culpa tempor anim et nulla exercitation.

          Et et est officia non eiusmod tempor dolor. Reprehenderit ipsum ut enim aliquip ipsum id aute ullamco dolor. Minim nulla pariatur Lorem nulla enim incididunt tempor exercitation adipisicing quis officia ex occaecat tempor.

          Non pariatur excepteur cupidatat esse elit elit consequat. Nostrud dolor commodo cupidatat eu sunt velit aute. Deserunt veniam excepteur tempor esse aliquip culpa laborum nulla amet est labore. Eiusmod non cillum non anim duis laborum incididunt cillum. Nisi adipisicing proident cillum adipisicing ipsum quis consectetur.

          Quis magna est enim aute nostrud nulla aliqua magna veniam exercitation culpa laboris do reprehenderit. Voluptate laborum anim laborum laborum elit aute. Aliqua ex commodo officia incididunt aliquip consequat. Exercitation cupidatat cillum minim proident cillum eu ex mollit veniam sit officia eiusmod. Dolor Lorem nulla sit aute ea officia. Commodo est ut ad fugiat quis nostrud.

          Enim sunt duis aute ex mollit nisi exercitation dolore reprehenderit. Occaecat minim ipsum fugiat dolor ut mollit cupidatat excepteur esse minim commodo aliqua. Nisi dolor nulla non excepteur elit dolore excepteur adipisicing commodo nulla est proident commodo sit.

          Aliqua voluptate excepteur excepteur duis aute veniam elit minim qui. Adipisicing sint pariatur reprehenderit officia minim deserunt anim. Dolore dolor elit dolore mollit sunt magna id. Nostrud dolore cupidatat duis voluptate qui ullamco enim et ullamco culpa dolor amet. Proident magna nulla ad Lorem duis excepteur enim esse ex velit dolore laboris. Pariatur aliquip exercitation proident dolore ad pariatur minim excepteur anim aute ipsum irure sunt consectetur.

          Enim duis id amet mollit excepteur amet qui pariatur aliqua mollit. Lorem commodo dolore nostrud eiusmod ipsum do eiusmod deserunt enim do officia. Adipisicing sint exercitation laborum sunt labore consequat aliquip sit cillum duis voluptate. Mollit officia magna ex aliquip qui nulla cillum ad magna magna. Proident qui laborum dolore nostrud reprehenderit esse commodo nostrud occaecat laborum commodo nulla. Deserunt minim eu ut deserunt. Sit commodo aute cillum mollit cillum velit.

          Aliquip velit ipsum cillum ut consequat sunt labore pariatur pariatur deserunt do cillum sit elit. Nulla pariatur pariatur pariatur cupidatat deserunt veniam in cillum quis ut in id adipisicing adipisicing. Consectetur ad et aliquip dolore ad laboris esse minim. Lorem consectetur pariatur exercitation sint culpa irure et labore.

          Proident non sit aliqua commodo est duis ipsum est est Lorem ex proident ad culpa. Ad exercitation deserunt qui cillum occaecat deserunt et laborum ad tempor. Qui tempor aliqua voluptate ex nisi tempor commodo cillum consequat ea cillum. Veniam sit ex anim pariatur occaecat qui adipisicing ex ut qui. Quis est in ad id ex eu enim anim exercitation. Sit dolor consequat exercitation adipisicing eiusmod aute labore mollit consectetur. Cupidatat dolor non nostrud duis eiusmod.

          Deserunt excepteur tempor tempor deserunt magna labore labore. Reprehenderit incididunt id enim ut ex ex. Veniam ea irure esse quis dolor elit amet. Occaecat voluptate aliqua veniam consequat et. Esse culpa in aute in cupidatat ad aliqua incididunt.
        </p>
      </section-wrap>
    </main>
  </div>
</template>

<script>
import MegaMenuAlt from '../MegaMenuAlt.vue';
import MegaMenuAltDocs from './mega-menu-alt.md';

export default {
  components: { MegaMenuAlt },
  readme: { custom: MegaMenuAltDocs, source: true, html: false },
  data() {
    return {
      primaryLinks: [
        { text: 'Home', href: '/home-page' },
        { text: 'Study at Melbourne', href: 'https://study.unimelb.edu.au/' },
        { text: 'Research and innovation', href: 'https://research.unimelb.edu.au/' },
        { text: 'Safety and respect', href: 'https://students.unimelb.edu.au/campus-life/policy-and-conduct/respect-at-the-university' },
        { text: 'Virtual Campus', href: 'https://study.unimelb.edu.au/discover/virtual-tour' },
      ],
      secondaryLinks: [
        { text: 'Current students', href: 'https://students.unimelb.edu.au/' },
        { text: 'Staff', href: 'https://staff.unimelb.edu.au/' },
        { text: 'Library', href: 'https://library.unimelb.edu.au/' },
        { text: 'Alumni', href: 'https://www.unimelb.edu.au/alumni' },
      ],
    };
  },
};
</script>
