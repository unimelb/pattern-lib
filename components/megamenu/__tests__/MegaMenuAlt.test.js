import { shallow } from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import MegaMenuAlt from '../MegaMenuAlt.vue';

expect.extend(toHaveNoViolations);

describe('MegaMenuAlt', () => {
  it('should match snapshot', () => {
    const result = shallow(MegaMenuAlt).element;

    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    // Set up our document body
    document.body.innerHTML = '<main id="main">'
        + '</main>';
    const html = shallow(MegaMenuAlt).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
