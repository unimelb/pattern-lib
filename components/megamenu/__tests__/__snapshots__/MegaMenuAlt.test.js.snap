// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MegaMenuAlt should match snapshot 1`] = `
<header
  class="mega-menu-alt"
>
  <a
    class="mega-menu-alt__skip-link screen-reader-jump-to__skippy screenreaders-only screenreaders-only-focusable btn--inverted btn btn--xsml"
    href="#main"
  >
    
    Skip to content
  
  </a>
   
  <div
    class="mega-menu-alt__inner bg-inverted"
  >
    <!---->
     
    <div
      class="mega-menu-alt__menu"
    >
      <div
        class="mega-menu-alt__menu-top"
      >
        <nav
          aria-label="Quick links"
        >
          <ul
            class="mega-menu-alt__secondary-links list-reset"
          >
            <li>
              <a
                class="link-reset"
                href="https://students.unimelb.edu.au/"
              >
                
                Current students
              
              </a>
            </li>
            <li>
              <a
                class="link-reset"
                href="https://staff.unimelb.edu.au/"
              >
                
                Staff
              
              </a>
            </li>
            <li>
              <a
                class="link-reset"
                href="https://library.unimelb.edu.au/"
              >
                
                Library
              
              </a>
            </li>
            <li>
              <a
                class="link-reset"
                href="https://www.unimelb.edu.au/alumni"
              >
                
                Alumni
              
              </a>
            </li>
          </ul>
        </nav>
         
        <ul
          class="mega-menu-alt__actions list-reset"
        >
          <li
            class="mega-menu-alt__contact"
          >
            <a
              class="link-reset"
              href="https://www.unimelb.edu.au/contact"
            >
              
              Contact
            
            </a>
          </li>
           
          <li
            class="mega-menu-alt__search"
          >
            <button
              aria-expanded="false"
              class="link-reset"
            >
              
              Search
              
              <svg
                focusable="false"
                role="presentation"
              >
                <use
                  xlink:href="#icon-search"
                />
              </svg>
            </button>
          </li>
           
          <li
            class="mega-menu-alt__mobile-menu"
          >
            <button
              aria-expanded="false"
              class="link-reset"
            >
              
              Menu
              
              <svg
                aria-label="icon menu"
                focusable="false"
                viewBox="10 10 26 28"
              >
                <path
                  d="M6 36h36v-4H6v4zm0-10h36v-4H6v4zm0-14v4h36v-4H6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
       
      <nav
        aria-label="Site"
      >
        <ul
          class="mega-menu-alt__primary-links list-reset"
        >
          <li>
            <a
              class="link-reset"
              href="/home-page"
            >
              
              Home
            
            </a>
          </li>
          <li>
            <a
              class="link-reset"
              href="https://study.unimelb.edu.au/"
            >
              
              Study at Melbourne
            
            </a>
          </li>
          <li>
            <a
              class="link-reset"
              href="https://research.unimelb.edu.au/"
            >
              
              Research and innovation
            
            </a>
          </li>
          <li>
            <a
              class="link-reset"
              href="https://students.unimelb.edu.au/campus-life/policy-and-conduct/respect-at-the-university"
            >
              
              Safety and respect
            
            </a>
          </li>
          <li>
            <a
              class="link-reset"
              href="https://study.unimelb.edu.au/discover/virtual-tour"
            >
              
              Virtual Campus
            
            </a>
          </li>
        </ul>
      </nav>
       
      <!---->
       
      <div
        class="page-header-search"
      >
        <!---->
      </div>
    </div>
  </div>
</header>
`;
